contracts/lido_terra_hub/src/contract.rs

```diff
@@ -213,8 +184,105 @@ pub fn execute(deps: DepsMut, env: Env, info: MessageInfo, msg: ExecuteMsg) -> S
-        ExecuteMsg::MigrateUnbondWaitList { limit: _ } => Err(StdError::generic_err("forbidden")),
+        ExecuteMsg::MigrateUnbondWaitList { limit } => {
+            migrate_unbond_wait_lists(deps.storage, limit)
+        }
+        ExecuteMsg::PauseContracts {} => execute_pause_contracts(deps, env, info),
+        ExecuteMsg::UnpauseContracts {} => execute_unpause_contracts(deps, env, info),
+        ExecuteMsg::AddGuardians { addresses } => execute_add_guardians(deps, env, info, addresses),
+        ExecuteMsg::RemoveGuardians { addresses } => {
+            execute_remove_guardians(deps, env, info, addresses)
+        }
+    }
+}
+
+pub fn execute_add_guardians(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    guardians: Vec<String>,
+) -> StdResult<Response> {
+    let config = CONFIG.load(deps.storage)?;
+    let owner = deps.api.addr_humanize(&config.creator)?;
+
+    if info.sender != owner {
+        return Err(StdError::generic_err("unauthorized"));
+    }
+
+    for guardian in &guardians {
+        GUARDIANS.save(deps.storage, guardian.clone(), &true)?;
+    }
+
+    Ok(Response::new()
+        .add_attributes(vec![attr("action", "add_guardians")])
+        .add_attributes(guardians.iter().map(|g| attr("value", g))))
+}
+
+pub fn execute_remove_guardians(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    guardians: Vec<String>,
+) -> StdResult<Response> {
+    let config = CONFIG.load(deps.storage)?;
+    let owner = deps.api.addr_humanize(&config.creator)?;
+
+    if info.sender != owner {
+        return Err(StdError::generic_err("unauthorized"));
     }
+
+    for guardian in &guardians {
+        GUARDIANS.remove(deps.storage, guardian.clone());
+    }
+
+    Ok(Response::new()
+        .add_attributes(vec![attr("action", "remove_guardians")])
+        .add_attributes(guardians.iter().map(|g| attr("value", g))))
+}
+
+pub fn execute_pause_contracts(deps: DepsMut, _env: Env, info: MessageInfo) -> StdResult<Response> {
+    let config = CONFIG.load(deps.storage)?;
+    let owner = deps.api.addr_humanize(&config.creator)?;
+
+    if !(info.sender == owner || GUARDIANS.has(deps.storage, info.sender.to_string())) {
+        return Err(StdError::generic_err("unauthorized"));
+    }
+
+    let mut params: Parameters = PARAMETERS.load(deps.storage)?;
+    params.paused = Some(true);
+
+    PARAMETERS.save(deps.storage, &params)?;
+
+    let res = Response::new().add_attributes(vec![attr("action", "pause_contracts")]);
+    Ok(res)
+}
+
+pub fn execute_unpause_contracts(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+) -> StdResult<Response> {
+    let config = CONFIG.load(deps.storage)?;
+    let owner = deps.api.addr_humanize(&config.creator)?;
+
+    if info.sender != owner {
+        return Err(StdError::generic_err("unauthorized"));
+    }
+
+    let old_unbond_wait_list_entries = read_old_unbond_wait_lists(deps.storage, Some(1u32))?;
```

Comment:
This PR will be brought to the mainnet way after the audited migration is going to happen: the old waitlists are going to be empty by that time. Do we really need this check here?

Reply:
All v1->v2 migration related logic is removed now